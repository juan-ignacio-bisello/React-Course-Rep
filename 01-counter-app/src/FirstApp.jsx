import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle } ) => {


    const newMessage = 'Ignacio';
    return ( 
        // <> funciona como un Fracmento sin la necesidad de hacer la importaci√≥n
        <>
            <h1>{ title }</h1>

            <p>{ subTitle }</p>
            
            <p>No se </p>
        </>
        
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    title:'No hay',
    subTitle:'No hay subtitulo'
}