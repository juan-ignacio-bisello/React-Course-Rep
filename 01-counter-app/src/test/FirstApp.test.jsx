/** * @jest-environment jsdom */

import { render } from "@testing-library/react"
import { FirstApp } from "../FirstApp"


describe('Puebas en FirstApp', () => {
    test('Debe hacer mach con el snapshot', () => { 
        
        const title = 'Hola, soy goku.';
        const { container } = render( <FirstApp title={ title } /> );

        expect( container ).toMatchSnapshot();
     })

     test('Debe mopstrar el titulo en un h1', () => { 
        
        const title = 'Hola soy goku';
        const { container, getByText } = render( <FirstApp title={ title } /> );

        expect( getByText( title ) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect( h1.innerHTML ).toContain( title );
      })
})

