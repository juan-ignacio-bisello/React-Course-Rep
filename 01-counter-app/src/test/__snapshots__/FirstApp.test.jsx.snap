// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en FirstApp Debe hacer mach con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy goku.
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No se 
  </p>
</div>
`;
